### PR TITLE
Derive bitcoin wallet

### PR DESCRIPTION
**Pull request by pkova**
As per the bounty at https://grants.urbit.org/bounties/794126751-bitcoin-key-derivation.

### CODE REVIEW

File: test/assets/wallet0.json [hunk 1]
```diff
@@ -23,7 +22,8 @@
    "private": "bc484a9d6056138675fd8f24e5c1bdda649c7acb3fcfbc9552fffcf14a20023f",
    "chain": "6544c8c43d875da4b3bb9871d0d2d1b8253dc6a1e2c1666bca865fe18543828b",
    "address": "0x9988685Fc1E5c4079AaEA6A405e0f28956b1bbBe"
-  }
+  },
+  "derivationPath": "m/44'/60'/0'/0/0"
```
Comment (pkova) [inline test/assets/wallet0.json:26]: Note that the derivation path is now per-wallet, not under `meta`. Breaking change.

**Comment by pkova** [comment]
Also, the wallet output references the UP8 HD Wallet spec at https://github.com/urbit/urbit-key-generation/blob/master/test/assets/wallet0.json#L7 (located at https://github.com/urbit/fora-posts/blob/master/proposals/posts/~2018.11.8..19.31.59..ba77~.md), which is not going the be accurate after this change. Do we need a new spec for this?

**Comment by galenwp** [comment]
exciting

**Comment by jtobin** [comment]
Two things come to mind:

* Per your comment, this library is, or at least has in practice become, the canonical JS implementation of UP8.  So it feels like it should implement the UP8 spec faithfully.

* It feels a little weird to add a specific bitcoin branch to the UP8 wallet itself.  But if we do, and update UP8 accordingly, then that has implications on other UP8 libraries that are floating around, e.g., in Hoon and Haskell.

  (Updating those certainly seems to be outside the scope of the grant deliverables, of course.)

After mulling it over, I think it's probably fine to just add the functionality here, i.e. without updating the spec, since this library doesn't *strictly* claim to be a precise no-more-and-no-less implementation of UP8.

If we do want to make a bitcoin branch a permanent fixture of the UP8 wallet, so that all proper UP8 wallet implementations must handle a bitcoin branch, then that can just be handled as a separate matter.

File: src/index.js [hunk 2]
```diff
@@ -20,10 +20,12 @@ const CHILD_SEED_TYPES = {
   SPAWN: 'spawn',
   VOTING: 'voting',
   MANAGEMENT: 'management',
-  NETWORK: 'network'
+  NETWORK: 'network',
+  BITCOIN: 'bitcoin'
 }
 
 const DERIVATION_PATH = "m/44'/60'/0'/0/0"
+const BTC_DERIVATION_PATH = "m/44'/0'/0'/0/0"
```
Comment (timlucmiptev) [inline src/index.js:28]: Is there are reason for using 44 instead of 84 addresses (segwit/Bech32)?

**Review by timlucmiptev — COMMENTED** [review]
Is there are reason for using 44 instead of 84 addresses (segwit/Bech32)?

File: src/index.js (hunk 2 above)
Comment (pkova) [reply src/index.js]: There isn't. Good catch.

**Comment by jalehman** [comment]
@jtobin it appears you've already given this a look, so formally requesting your review just to get this moving towards a merge.

File: src/index.js [hunk 3]
```diff
@@ -203,17 +206,19 @@ const deriveNodeKeys = (mnemonic, passphrase) => {
  *
  * @param  {Uint8Array}  master a master seed
  * @param  {String}  type one of 'ownership', 'transfer', 'spawn', 'voting',
- *   'management'
+ *   'management', 'bitcoin'
```
Comment (Fang-) [inline src/index.js:209]:
```suggestion
 *                   'management', 'bitcoin'
```

**Review by Fang- — APPROVED** [review]
I think this is all good. Fairly small change. The breaking part seems sane regardless of the goal of this PR, under the assumption that this is "key generation that supports UP8", not "exclusively UP8". Keeping UP8 as a _primary_ use case seems fair, and this doesn't diminish that.

Updating the `/lib/keygen.hoon` to match might be nice, but it's probably not gonna see serious Bitcoin-related use as early as this JS version will.

File: src/index.js (hunk 3 above)
Comment (Fang-) [reply src/index.js:209]: Realizing now this was already ugly prior to your changes, just like the rest of these doccomments. Do as you wish!

**Comment by pkova** [comment]
My initial mental model regarding this was that we just generate a single bitcoin address like we do for the ethereum stuff. In actuality we're gonna need a real bip32 HD wallet here, but luckily we're dealing with HD wallets already as per the UP8 spec.

Changed the derivation path to the more appropriate `m/84'/0'/0'` for this reason.

File: src/index.js (hunk 3 above)
Comment (pkova) [reply src/index.js]: Fixed them all

**Review by jtobin — APPROVED** [review]
LGTM!

**Pull request merged**